File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @roll = Hash.new { [] }
  end

  def students(grade)
    @roll[grade]
  end

  def add(name, grade)
    (@roll[grade] <<= name).sort!
  end

  def students_by_grade
    @roll.sort.map { { grade: _1, students: _2 } }
  end
end

File: javascript/grade-school/grade-school.js
class School {
  constructor(){
    this.students = {};
  }

  add(student, grade){
    if (this.students[grade]){
      this.students[grade].push(student);
      this.students[grade].sort();
    }
    else {
      this.students[grade] = [student];
    }
  }

  grade(level){
    return this.students[level] || [];
  }

  roster(){
    return this.students;
  }
}

module.exports = School;
